### PR TITLE
[Stashing] Replace ListRow component

### LINKED ISSUE #7275: [Stashing] Replace the usage of the ListRow component with a custom one
This issue was brought up in review https://github.com/desktop/desktop/pull/7210#discussion_r272649999. 

I think you could crimp the selection logic and click events from inside `ListRow` without needing to change the behaviour of `ListRow`. Looking at the setup, there's a lot of hard-coded values because of what `ListRow` is expecting:

```jsx
<ListRow
  rowCount={1}                             // necessary?
  rowIndex={0}                              // necessary?
  selectable={true}                       // necessary?
  selected={this.props.isShowingStashEntry} 
  onRowClick={this.onStashEntryClicked}
  tabIndex={0}                               // necessary?
  style={StashListRowStyle}        // pass in rowHeight directly?
  className="stash-entry-row" // could live inside new component
  >
```

I'd love to see an exploration of alternatives here because I think you could get away with something simpler than `ListRow` for your use case.

_Originally posted by @shiftkey_

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
Fixes #7275

This PR can be merged after we release stashing to beta. Note that this is based on `stash-changes-on-branch` to make the diff clearer. Once #7210 is merged, i'll update the base to `development`.

Addresses feedback left in this https://github.com/desktop/desktop/pull/7210#discussion_r274048208.

### CODE REVIEW

File: app/src/ui/changes/changes-list.tsx [hunk 1]
```diff
@@ -36,10 +36,10 @@ import { ICommitContext } from '../../models/commit'
 import { RebaseConflictState } from '../../lib/app-state'
 import { ContinueRebase } from './continue-rebase'
 import { enablePullWithRebase } from '../../lib/feature-flag'
-import { ListRow } from '../lib/list/list-row'
 import { Octicon, OcticonSymbol } from '../octicons'
 import { FocusContainer } from '../lib/focus-container'
 import { IStashEntry } from '../../models/stash-entry'
+import classNames = require('classnames')
```
Comment (shiftkey) [inline app/src/ui/changes/changes-list.tsx]:
```suggestion
import * as classNames from 'classnames'
```

Comment (iAmWillShepherd) [reply app/src/ui/changes/changes-list.tsx]: I have no clue why Vs Code did that style of import...

Comment (shiftkey) [reply app/src/ui/changes/changes-list.tsx]: My gut feeling it's something related to the TS services in VSCode, and `require` is a safer bet for Node projects...

**Comment by outofambit** [comment]
@iAmWillShepherd i think this needs to be updated with the changes made in #7313 (which is now on development) before we can review and merge.

**Pull request merged**